Sources/Fullscreen/FrontpageSearch/FrontpageSearchView.swift

```diff
@@ -0,0 +1,221 @@
+        let footer = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: footerSize,
+                                                                 elementKind: UICollectionView.elementKindSectionFooter,
+                                                                 alignment: .bottom
+        )
+        let section = NSCollectionLayoutSection(group: group)
+        section.boundarySupplementaryItems = [header, footer]
+        section.contentInsets = NSDirectionalEdgeInsets(
+            vertical: 0,
+            horizontal: Self.horizontalSpacing
+        )
+        return section
+    }
+
+    private typealias DataSource = UICollectionViewDiffableDataSource<FrontpageSearchSection, FrontpageSearchGroupItem>
+    private lazy var dataSource: DataSource = {
+        let dataSource = DataSource(
+            collectionView: collectionView,
+            cellProvider: { (collectionView, indexPath, item) -> UICollectionViewCell? in
+                switch item.groupType {
+                case .searchResult:
+                    switch item.displayType {
+                    case .standard:
+                        let cell = collectionView.dequeue(FrontpageSearchResultCell.self, for: indexPath)
+                        cell.configure(with: item)
+                        return cell
+                    case .recommend, .myFindings, .companyProfile, .geo:
+                        let cell = collectionView.dequeue(FrontpageSearchImageResultCollectionViewCell.self, for: indexPath)
+                        cell.configure(with: item, remoteImageViewDataSource: self.remoteImageViewDataSource)
+                        cell.delegate = self.delegate
+                        return cell
+                    case .myFindingsList:
+                        // should return an image cell with a circular image - not yet implemented
+                        let cell = collectionView.dequeue(FrontpageSearchImageResultCollectionViewCell.self, for: indexPath)
+                        cell.configure(with: item, remoteImageViewDataSource: self.remoteImageViewDataSource)
+                        return cell
+                    }
+                case .locationPermission:
+                    let cell = collectionView.dequeue(FrontpageSearchLocationPermissionCell.self, for: indexPath)
+                    cell.configure(with: item.title)
+                    cell.delegate = self.delegate
+                    return cell
+                case .showMoreResults:
+                    let cell = collectionView.dequeue(FrontpageSearchMoreResultsCollectionViewCell.self, for: indexPath)
+                    cell.configure(with: item.title)
+                    return cell
+                }
+            }
+        )
+
+        dataSource.supplementaryViewProvider = { [weak self] collectionView, kind, indexPath in
+            if kind == "UICollectionElementKindSectionHeader" {
```

Comment:
It's probably better to compare against `UICollectionView.elementKindSectionHeader` here, and `UICollectionView.elementKindSectionFooter` below

Reply:
Yeah, that looks better too.